app.rb

```diff
@@ -0,0 +1,60 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'book'
+class App
+  attr_accessor :books, :peoples, :rentals
+
+  def initialize
+    @books = []
+    @peoples = []
+    @rentals = []
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books available'
+    else
+      @books.each { |book| puts "#{book.title} by #{book.author}" }
+    end
+  end
+
+  def list_people
+    if @peoples.empty?
+      puts 'No people available'
+    else
+      @peoples.each { |person| puts "ID: #{person.id} , Name: #{person.name}, Age: #{person.age}" }
+    end
+  end
+
+  def create_student(age, has_parent_permission, name)
+    student = Student.new(age, parent_permission: has_parent_permission, name: name)
+    @peoples << student
+  end
+
+  def create_teacher(age, specialization, name)
+    teacher = Teacher.new(age, specialization, name: name)
+    @peoples << teacher
+  end
+
+  def create_book(title, author)
+    book = Book.new(title, author)
+    @books << book
+  end
+
+  def create_rental(date, book, person)
+    rental = Rental.new(date, book, person)
+    @rentals << rental
+  end
+
+  def list_rentals(person_id)
+    person = @peoples.find { |p| p.object_id == person_id }
+    if person.nil?
+      puts 'Person not found'
+    else
+      rentals = @rentals.select { |r| r.person.object_id == person_id }
+      rentals.each { |rental| puts "#{rental.book.title} by #{rental.book.author}" }
+    end
+  end
```

Comment:
- [ ] Also for `Rental`, once I wanted to create a Rental based on `Book` and `Person` , It was thrown the following exception 😢 , Kindly fix that and ensure the creation of a `Rental` is enabled 👍  

![Capture](https://github.com/rbd3/School-Library/assets/80458199/62a6d28c-38f0-4afb-89e5-29b6c630d277)